wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -422,6 +433,30 @@ pub async fn handle_wallet_command(
             Ok(ConsoleCommand::Print("Success".to_owned()))
         }
 
+        WalletCommand::CreateNewAccount => {
+            let new_account_index = controller_opt
+                .as_mut()
+                .ok_or(WalletCliError::NoWallet)?
+                .create_account()
+                .map_err(WalletCliError::Controller)?;
+            Ok(ConsoleCommand::Print(format!(
+                "Success, the new account index is: {}",
+                new_account_index
+            )))
+        }
+
+        WalletCommand::SelectAccount { account_index } => {
+            controller_opt
+                .as_mut()
+                .ok_or(WalletCliError::NoWallet)?
+                .select_account(account_index)
+                .map_err(WalletCliError::Controller)?;
+            Ok(ConsoleCommand::SetStatus {
+                status: format!(" ({})", account_index),
+                print: format!("Success, the selected account index is: {}", account_index),
+            })
```

Comment:
Looks like the status line is updated with the account number when the `SelectAccount` command is issued. Is the account number displayed if there are multiple accounts, regardless whether the select account command was run?

Reply:
I think it is missing an if account_index == DEFAULT then set the status to "" 
I think that is what @TheQuantumPhysicist mentioned to only show account if it is not the default one, not sure?

Reply:
In general, it seems quite error prone to manually set the status string every time it's supposed to change. Is it possible to somehow generate the status string on demand when it's being displayed? It would need the access to the wallet state to check current account and the total number of active accounts.

Reply:
that was my initial plan, but the prompt and the wallet communicate through channels in separate threads.

Reply:
As long as this variable is trapped in the CLI... I'm good with it. It shouldn't touch the controller though.

Reply:
Moved it to the WalletPrompt, but not sure what to do with non interactive modes, do I create another enum with commands with account_index, or add optional account_index in the existing ones and use it or if empty the selected one?

Reply:
discussed with @pavel-kokolemin, I will make the non interactive modes work the same as the interactive one with a state of the selected account index.